src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphDAO.scala

```diff
@@ -92,4 +98,94 @@ trait GraphDAO {
   }
+
+  import scala.reflect.runtime.universe._
+  import scala.reflect.runtime.{universe=>ru}
+  import scala.annotation.meta.field
+
+  /**
+   * Takes the fields of obj annotated with VertexProperty and sets them on vertex.
+   * Vertex property keys are prefixed with an underscore.
+   *
+   * @param obj source of data with which to update the vertex
+   * @param vertex the vertex to update
+   * @tparam T the type of obj
+   * @return the updated vertex
+   */
+  def setVertexProperties[T: TypeTag: ClassTag](obj: T, vertex: Vertex): Vertex = {
+
+    def getProperties(obj: T, prefix: String): Iterable[(String, Option[Any])] = {
+      val mirror = ru.runtimeMirror(obj.getClass.getClassLoader)
+
+      ru.typeTag[T].tpe.members.map { member =>
+        if ((member.asTerm.isVal || member.asTerm.isVar) && member.annotations.find(_.tree.tpe.typeSymbol == ru.typeOf[VertexProperty@field].typeSymbol).isDefined) {
+          val fieldMirror = mirror.reflect(obj).reflectField(member.asTerm)
+          fieldMirror.get match {
+            case None => vertex.getPropertyKeys.filter(_.startsWith(prefix)).map(_ -> None)
+            case Some(v) => Seq(prefix + member.name.decodedName.toString.trim -> Option(v))
+            case v => Seq(prefix + member.name.decodedName.toString.trim -> Option(v))
+          }
+        } else {
+          Seq.empty
+        }
+      } flatten
+    }
+
+    getProperties(obj, "_").foreach(_ match {
+      case (key, None) => vertex.removeProperty(key)
+      case (key, Some(value: DateTime)) => vertex.setProperty(key, value.toDate)
+      case (key, Some(value)) => vertex.setProperty(key, value)
+    })
+
+    vertex.setProperty("_clazz", obj.getClass.getSimpleName)
+    vertex
+  }
+
+  /**
+   * Convert a graph vertex to an instance of T
+   * @param vertex
+   * @param otherProperties properties required to construct T but are not in the vertex. Keys must match the
+   *                        names of the fields on T to set
+   * @tparam T
+   * @return an instance of T populated from the vertex and otherProperties
+   */
+  def fromVertex[T: TypeTag](vertex: Vertex, otherProperties: Map[String, Any]): T = {
+    val vertexProperties: Map[String, Any] = vertex.getPropertyKeys.map { key => key -> vertex.getProperty(key) }.toMap
+    fromPropertyMap(otherProperties ++ vertexProperties)
+  }
+
+  /**
+   * Constructs and instance of T given a map of properties where the keys correspond to the name of the fields in T
+   * (with or without a leading underscore)
+   *
+   * @param vertexProperties
+   * @tparam T
+   * @return an instance of T populated from vertexProperties
+   */
+  def fromPropertyMap[T: TypeTag](vertexProperties: Map[String, Any]): T = {
+    val classT = ru.typeOf[T].typeSymbol.asClass
+    val classMirror = ru.runtimeMirror(getClass.getClassLoader).reflectClass(classT)
+    val ctor = ru.typeOf[T].decl(ru.termNames.CONSTRUCTOR).asMethod
+    val ctorMirror = classMirror.reflectConstructor(ctor)
```

Comment:
technically, I think there can only ever be 1 constructor for any scala class. The "overloaded" methods are just short cuts that must call the 1 true constructor.